@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -34,91 +31,86 @@ const neutralColor = new THREE.Color(1, 1, 1);
 const tmpIconColor = new THREE.Color();
 
 /**
- * The `PoiRenderBufferBatch` contains the geometry and the material for all POIs that share the
- * same icon image ({@link @here/harp-datasource-protocol#ImageTexture}).
- *
- * @remarks
- * If the image is the same, all the objects in this batch can
- * share the same material, which makes them renderable in the same draw call, whatever the number
- * of actual objects (WebGL limits apply!).
- *
- * There is a `PoiRenderBufferBatch` for every icon in a texture atlas, since the size of the icon
- * in the atlas as well as the texture coordinates are specified in the `PoiRenderBufferBatch`.
+ * @internal
+ * Buffer for POIs sharing same material and render order, renderable in a single draw call
+ * (WebGL limits apply, see {@link BoxBuffer}).
  */
-class PoiRenderBufferBatch {
-    // Enable trilinear filtering to reduce flickering due to distance scaling
-    static trilinear: boolean = true;
-
-    boxBuffer: BoxBuffer | undefined;
-
-    private m_material?: THREE.Material | THREE.Material[];
+export class PoiBuffer {
+    private m_refCount: number = 0;
 
     /**
-     * Create the `PoiRenderBufferBatch`.
-     *
-     * @param mapView - The {@link MapView} instance.
-     * @param scene - The three.js scene to add the POIs to.
-     * @param imageItem - The icon that will have his material shared.
-     * @param renderOrder - RenderOrder of the batch geometry's [[Mesh]].
+     * Creates a `PoiBuffer`
+     * @param buffer -
+     * @param layer - The {@link TextCanvas} layer used to render the POIs.
      */
     constructor(
-        readonly mapView: MapView,
-        readonly scene: THREE.Scene,
-        readonly imageItem: ImageItem,
-        readonly renderOrder: number
+        readonly buffer: BoxBuffer,
+        readonly layer: TextCanvasLayer,
+        private readonly m_onDispose: () => void
```

Comment:
this could be optional I guess?

Reply:
This class is only meant to be instantiated by PoiBatch in the same file, which always provides a dispose function. PoiBatch is exported only because PoiInfo calls to decreaseRefCount when destroyed.

I could add a IPoiBatch interface with the ref count functions and just export that to make it clearer. But PoiBatch only has those public functions so I don't think it's worth it.

Reply:
ok, understood.